### PR TITLE
Use bytes.hex() APIs

### PR DESCRIPTION
**Pull request by dholth**
### Description

Use a few more modern hex APIs; use set for "in" queries.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda_content_trust/signing.py [hunk 1]
```diff
@@ -74,7 +73,7 @@ def wrap_as_signable(obj):
     Raises ❌TypeError if the given object is not a JSON-serializable type per
     SUPPORTED_SERIALIZABLE_TYPES
     """
-    if not type(obj) in SUPPORTED_SERIALIZABLE_TYPES:
+    if not isinstance(obj, SUPPORTED_SERIALIZABLE_TYPES):
```
Comment (dholth) [inline conda_content_trust/signing.py:76]: I would want to show this error after calling json.dumps() not as a pre-check.

Comment (kenodegard) [reply conda_content_trust/signing.py:76]: yeah I've looked at this before, we don't use `isinstance` here because we don't want to support subclasses

Comment (dholth) [reply conda_content_trust/signing.py]: Changed back. Should switch to a dumps/loads or restructure the way this function works - this check is only testing the outer level container anyway so a list of object() would pass the check, but fail json.dumps()

**Comment by jezdez** [comment]
pre-commit.ci autofix

File: conda_content_trust/common.py [hunk 2]
```diff
@@ -72,7 +72,7 @@
 # The only types we're allowed to wrap as "signables" and sign are
 # the JSON-serializable types.  (There are further constraints to what is
 # JSON-serializable in addition to these type constraints.)
-SUPPORTED_SERIALIZABLE_TYPES = set((dict, list, tuple, str, int, float, bool, type(None)))
+SUPPORTED_SERIALIZABLE_TYPES = {dict, list, tuple, str, int, float, bool, type(None)}
```
[the remark this answers is not in the document]
Comment (jezdez) [reply conda_content_trust/common.py:75]: Robots overruled our code style

**Review by beeankha — APPROVED** [review]

**Review by kenodegard — APPROVED** [review]

**Pull request merged**